src/robotFactory.js

```diff
@@ -1,10 +1,75 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
```

Comment:
```suggestion
  constructor(
  name, 
  weight, 
  coords, 
  chipVersion,
) {
```
Now your args looks much better